### ISSUE TITLE
Default created storeview sales sequences are missing a prefix

### ISSUE THREAD

**Issue by hostep** [magento/magento2#30863]
### Preconditions (*)

Magento 2.4-develop

### Steps to reproduce (*)
1. Setup a Magento 2 shop (doesn't matter which version, this has been bugged since forever)
2. Look at the `sales_sequence_profile` table in the database.

### Expected result (*)
1. The profiles associated to storeview ID 1 should have as prefix `1`

### Actual result (*)
1. The profiles associated to storeview ID 1 have as prefix `NULL`

![DB prefix](https://user-images.githubusercontent.com/51679138/98948208-d2b06200-24fe-11eb-95a5-f663d0ce5dbe.png)

### Discussion
When creating a second storeview, it gets prefix `2`, so one would expect that the default storeview which is created when setting up a shop gets prefix `1`.

When this gets fixed, it should **only** be fixed for new Magento shops being setup, please don't add a db migration script to fix it for existing shops, that would be very unexpected for shop owners.

### Workaround
The following SQL can be executed as a workaround if you want the prefix to be set correctly/consistently with other storeviews:

```sql
UPDATE `sales_sequence_profile`
SET `prefix` = 1
WHERE `profile_id` IN
    (SELECT `meta_id`
     FROM `sales_sequence_meta`
     WHERE `store_id` = 1);
```

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Alfa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-39058` were created

**Issue Available**: @engcom-Alfa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by vlmed** [comment]
@magento I am working on this

**Comment by sivaschenko** [comment]
@hostep agree, priority increased to P2. However, it's still currently targeting 2.5 due to BIC nature of the fix

**Comment by brohon** [comment]
I just ran into this issue in 2024. 

I created a new store, but the rows in sales_sequence_profile and sales_sequence_meta were not created. 

On my development and staging environments, they were created, and I can see there are four extra rows on those tables in those enviroments.

**Comment by engcom-Delta** [comment]
Hi @hostep ,

Thanks for your reporting and collaboration.
We have tried to reproduce the issue in Latest 2.4-develop instance and the issue is still reproducible. Kindly refer the screenshots.

**Steps to reproduce**
1. Install Magento
2. Look at the sales_sequence_profile table in the database.
3. Observe, The profiles associated to storeview ID 1 have as prefix NULL

![Image](https://github.com/user-attachments/assets/f4fc46bd-c13e-4312-9c11-413b41f7300d)

Hence Confirming the issue.

Thanks.
